service/controller/app/v1/resource/chartcrd/create.go

```diff
@@ -0,0 +1,61 @@
+package chartcrd
+
+import (
+	"context"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/controllercontext"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	secretName := key.SecretName(cr)
+
+	if secretName != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring chart crd creation on tenant cluster")
+
+		ctlCtx, err := controllercontext.FromContext(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		k8sExtClient, err := apiextensionsclient.NewForConfig(ctlCtx.RESTConfig)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c := k8scrdclient.Config{
+			K8sExtClient: k8sExtClient,
+			Logger:       r.logger,
+		}
+
+		crdClient, err := k8scrdclient.New(c)
+		crdBackoff := backoff.NewMaxRetries(3, 1*time.Second)
+
+		err = crdClient.EnsureCreated(ctx, v1alpha1.NewChartCRD(), crdBackoff)
+		if err != nil {
+			r.logger.LogCtx(ctx, "level", "debug", "error", "failed to ensured chart crd creation on tenant cluster")
```

Comment:
I think logging and returning an error at the same time is an anti-pattern. The returned error will be logged at the end anyway and we do not need it twice. I would like to drop that line.